### PR TITLE
Spectating Marine and Xeno huds

### PR DESCRIPTION
**Pull request by coolmankid12345**
## About the PR

Observer mains rejoice! You can now see the job icons on marines that have them and the health and plasma of Xenos. Two new actions are added as a ghost where you can toggle both on and off. (Icons are placeholder as I don't have any actual ones and I'm too lazy to look, I was gonna add health icons for marines but it wasn't really working so I scrapped it) 
## Why / Balance

I love seeing the squad icons and it always feel like I am missing something when spectating so now we can see them. Xenos wasn't gonna be included but I added it too even though I don't like xenos.
## Technical details

Now this code could definitely be better but we still figuring it out again. Definitely some scuff shit in here but I hope you can take a look. 

Adds two new components called `CMGhostMarineHudComponent` and `CMGhostXenoHudComponent` which are added and removed from the two new actions. There are two `CMGhostSystems` in the server and the client to handle the removing/adding of actions and components.
## Media

https://github.com/user-attachments/assets/115da041-6f36-4002-a4ec-3048cc1e4e3e
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Marine icons and Xeno plasma/health for spectators to see

### CODE REVIEW

**Comment by Kadeo64** [comment]
re-fucking-joice

File: Content.Client/_RMC14/Marines/MarineOverlay.cs [hunk 1]
```diff
@@ -36,8 +37,9 @@ public MarineOverlay()
 
     protected override void Draw(in OverlayDrawArgs args)
     {
-        if (!_entity.HasComponent<MarineComponent>(_players.LocalEntity))
-            return;
+        if (!_entity.HasComponent<CMGhostMarineHudComponent>(_players.LocalEntity))
+            if (!_entity.HasComponent<MarineComponent>(_players.LocalEntity))
+                return;
```
Comment (DrSmugleaf) [inline Content.Client/_RMC14/Marines/MarineOverlay.cs:42]: These two if statements can be merged 
```suggestion
        if (!_entity.HasComponent<CMGhostMarineHudComponent>(_players.LocalEntity) &&
            !_entity.HasComponent<MarineComponent>(_players.LocalEntity))
            {
                return;
            }
```

File: Content.Server/_RMC14/Mobs/CMGhostSystem.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public sealed class CMGhostSystem : EntitySystem
+    {
+        [Dependency] private readonly SharedActionsSystem _actions = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            //This shit is so scuffed but honest to god not sure what else I can use that isn't a duplicate
+            SubscribeLocalEvent<GhostHearingComponent, ComponentStartup>(OnGhostStartup);
```
Comment (DrSmugleaf) [inline Content.Server/_RMC14/Mobs/CMGhostSystem.cs:17]: We should probably have an RMCMobObserver but for now this is fine

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**